### DISCUSSION TITLE
How to subscribe SNS topics to an existing chatbot Slack configuration?

### DISCUSSION THREAD

**Discussion by debnon** [aws/aws-cdk#23600] [Q&A]
Hi. I have 2 stacks, each of which contain SNS topics which I would like the produce alerts on a single slack channel. 

This is easy to do with 1 stack, since I can just create the chatbot Slack configuration with the desired topics in that stack. However, I can't figure out how to subscribe additional topics to this same chatbot Slack configuration to make it work with 2. 

I'm using CDKv1. Any advice would be appreciated, thank you.

**Comment by peterwoodworth** [comment]
Which constructs are you currently using to configure your chatbot and SNS topics? A snippet of your code involving these constructs would be helpful 🙂

**Reply by debnon** [reply]
Hi. 

This is how the topics are created in both stacks:
```typescript
const snsTopic = new sns.Topic(this, `id1`, {
	displayName: "topic1",
	fifo: false,
});

errorRateAlarm.addAlarmAction(new cloudwatchActions.SnsAction(snsTopic));
```

This is how we currently subscribe the SNS topic of 1 stack:
```typescript
new chatbot.SlackChannelConfiguration(this, "id", {
	slackChannelConfigurationName: name,
	slackWorkspaceId: workspaceId,
	slackChannelId: slackchannelId,
	notificationTopics: [snsTopic],
});
```
 However we cannot do this in both stacks as you can only have 1 configuration for a given slack channel, and there doesn't seem to be an easy way to subscribe a new SNS topic to this config in a different stack 🤔 


Thanks for the help!

**Reply by peterwoodworth** [reply]
Your `SlackChannelConfiguration` construct is going to be the one in control of subscribing each topic, because the underlying CloudFormation resource is the one doing the work. You'll need some way to communicate to this construct (not an import) that there is a new topic, or else we may have to go the custom resource route.

One way you can avoid custom resources is to refactor your application such that both of these stacks are controlled by the same application. This way you can make use of cross-stack references, where your SlackChannelConfiguration stack depends on all other stacks where topics to subscribe to this chatbot are defined. You can then pass in topics to the stack through a prop, and set these passed in topics as notification topics for the SlackChannelConfiguration construct.

Another way you can avoid custom resources is to modify your SlackChannelConfiguration stack to [import each additional new topic](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_sns.Topic.html#static-fromwbrtopicwbrarnscope-id-topicarn) you create. Then you can pass in these imported topics directly to the SlackChannelConfiguration construct.
